docs/source/cli/model-training-inference/distributed/sagemaker.rst

```diff
@@ -213,6 +213,80 @@ from ``${DATASET_S3_PATH}`` as input and create a DistDGL graph with
 ``${NUM_PARTITIONS}`` under the output path, ``${OUTPUT_PATH}``.
 Currently we only support ``random`` as the partitioning algorithm.
 
+Launch hyper-parameter optimization task
+````````````````````````````````````````
+
+GraphStorm supports `automatic model tuning <https://docs.aws.amazon.com/sagemaker/latest/dg/automatic-model-tuning.html>`_
+with SageMaker AI,
+which allows you to optimize the hyper-parameters
+of your model with an easy-to-use interface.
+
+The ``launch/launch_hyperparameter_tuning.py`` script can act as a thin
+wrapper for SageMaker's `HyperParameterTuner <https://sagemaker.readthedocs.io/en/stable/api/training/tuner.html>`_.
+
+You define the hyper-parameters of interest by passing a python dictionary as a string,
+where the keys of the dictionary are the names of the parameters to tune,
+and each value is a dictionary that defines the type and tuning range
+of the parameter.
+
+.. code:: bash
+
+    # Example hyper-parameter ranges
+    python launch/launch_hyperparameter_tuning.py \
+        --hyperparameter-ranges '{"lr": {"type": "continuous", "min": 1e-5, "max": 1e-2, "scaling_type": "Auto"}, "num_layers": {"type": "integer", "min": 1, "max": 3}, "model_encoder_type": {"type": "categorical", "values": ["rgcn", "rgat"]}}'
+
+The supported parameter types are ``continuous``
+for real-values parameters, ``integer`` for numerical integer parameters,
+and ``categorical`` for discrete-value string parameters.
+These directly correspond to Sagemaker's
+`Dynamic hyper-parameters <https://docs.aws.amazon.com/sagemaker/latest/dg/automatic-model-tuning-define-ranges.html#automatic-model-tuning-define-ranges-dynamic>`~
+
+Continuous and integer parameters need to define a ``"min"`` and ``"max"`` value to use
```

Comment:
I don't think we need L244-L246. Users can refer to the SageMaker doc for more information.
Otherwise, we have to track the changes from SageMaker.

Reply:
Ah good point. Right now we use custom parsing of the ranges, i.e. our expected JSON syntax does not match SageMaker 1 to 1. 

Better if we instead directly use SageMaker's format and refer the users to that.